### ISSUE TITLE
Process.THREAD_PRIORITY_BACKGROUND affects performance

### ISSUE THREAD

**Issue by jiahaog** [tekartik/sqflite#738]
Hi, we're seeing some performance issues regarding the use of `Process.THREAD_PRIORITY_BACKGROUND` for the current Android implementation of the plugin.

https://github.com/tekartik/sqflite/blob/81c0c30aa59d7d3f229bf6ce0494458f96f893c2/sqflite/android/src/main/java/com/tekartik/sqflite/SqflitePlugin.java#L74

![Screen Shot 2021-12-03 at 15 38 10](https://user-images.githubusercontent.com/7111741/144570266-464c7186-dc3d-4b1e-83b4-48551f92a0a0.png)

In the trace above, we see that the sqflite thread is actually "Runnable" but not scheduled to run for a large portion of the time. This happens at a point of startup where everything is waiting on this method channel, which negatively affects startup time. In our benchmarks, we noticed that changing it to `Process.THREAD_PRIORITY_DEFAULT` or `Process.THREAD_PRIORITY_FOREGROUND` causes a decrease in startup latency by about 150-200ms. 

I understand that there's [a way](https://github.com/tekartik/sqflite/issues/459#issuecomment-645152781) to configure this priority from Dart, but would you be willing to share more context about the current default? Should we consider changing it to `Process.THREAD_PRIORITY_DEFAULT` instead? Thanks a lot!

**Comment by alextekartik** [comment]
I don't have a strong opinion about it. I made some quick experiment at the beginning (hence the ability to change the priority) and did not notice big differences but it was in a test context which is never as big as a real application context.

I guess I made the choice assuming that a background thread should have a lower priority than the main thread so that it does not impact Flutter UI. Personally I'm ok if the database layer is not super fast (it is always faster than fetching from the network), the UI should handle this gracefully. 

> Should we consider changing it to Process.THREAD_PRIORITY_DEFAULT instead? Thanks a lot!

Your investigation is very helpful. If you have a significant gain (and it seems so) and that it does not affect UI performance I would be totally ok to change it.

**Comment by jiahaog** [comment]
Thanks, I have opened https://github.com/tekartik/sqflite/pull/740 to change the default then 🙂

**Comment by alextekartik** [comment]
Thank you @jiahaog Published as 2.0.1

**Comment by jiahaog** [comment]
Thank you!
